csrc/evaluator_common.cpp

```diff
@@ -165,12 +165,13 @@ void PrecomputedValues::bindInputs(const KernelArgumentHolder& args) {
   for (const auto i : c10::irange((int64_t)inputs.size())) {
     const auto input = inputs[i];
     TORCH_INTERNAL_ASSERT(input != nullptr);
-    bindValue(input->evaluatorIndex(), *args[i]);
     if (auto tensor_input = dynamic_cast<TensorView*>(input)) {
       const auto& tensor = args[i]->as<at::Tensor>();
       if (!tensor.is_cpu()) {
         bindTensorMetaData(tensor_input, tensor);
       }
+    } else {
+      bindValue(input->evaluatorIndex(), *args[i]);
```

Comment:
This is only applied to non-tensor data, so no lifetime change should occur. Correct?

Reply:
Yes, exactly. This is the latest updated version. Before the last commit, I was doing `bindValue(input->evaluatorIndex(), *args[i]);` regardless of the type, so tensor objects always got stored.